fuel-merkle/src/binary/mrc.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{
+    binary::{
+        empty_sum,
+        Node,
+    },
+    common::Bytes32,
+};
+
+use crate::alloc::borrow::ToOwned;
+use alloc::vec::Vec;
+
+#[derive(Default, Debug, Clone)]
+pub struct MerkleRootCalculator {
+    stack: Vec<Node>,
+}
+
+impl MerkleRootCalculator {
+    pub fn new() -> Self {
+        Self { stack: Vec::new() }
+    }
+
+    pub fn push(&mut self, data: &[u8]) {
+        let node = Node::create_leaf(0, data);
+        self.stack.push(node);
+
+        while let Some(top_node) = self.stack.last() {
+            if self.stack.len() > 1 {
+                if let Some(second_top_node) = self.stack.get(self.stack.len() - 2) {
```

Comment:
```suggestion
        while self.stack.len() > 1 {
            let top_node = self.stack[self.stack.len() - 1];
            let second_top_node = self.stack[self.stack.len() - 2];
```

It seems we can also remove too `if` statements=)